zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -314,6 +324,17 @@ cdef _array_for_column_impl(object dtype,
             else:
                 adjustment_list = <list> adjustment_list_ptr
 
+            if asof_ix == 0:
+                # If the asof_ix == 0, this value may fall before the start of
+                # our window. In this case, we only want to apply data that is
+                # more recent than the current asof date stored at index 0.
+                with cython.boundscheck(False), cython.wraparound(False):
+                    asof_date = asof_dates[n]
+                    if asof_date >= index_0_asof_date_by_column_ix[column_ix]:
```

Comment:
I might comment here that this is `>=` rather than `>` because if we have the same asof but a newer timestamp this value should take precedence.

Reply:
done